File: Data Structures & Algorithms/validate-parentheses/submission-0.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> bucket;
        unordered_map<char, char> map;
        map['}'] = '{';
        map[']'] = '[';
        map[')'] = '(';

        for(char ch : s){
            if(map.find(ch) != map.end()){
                if(bucket.empty()){
                    return false;
                }

                char top = bucket.top();
                bucket.pop();

                if(top != map[ch]){
                    return false;
                }
            } else {
                bucket.push(ch);
            }
        }

        return bucket.empty();
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-0.js
class Solution {
    /**
     * @param {number[]} temperatures
     * @return {number[]}
     */
    dailyTemperatures(temperatures) {
        let stack = []
        let ans = Array(temperatures.length).fill(0)

        for(let i = 0; i < temperatures.length; i++){
            
            while(stack.length !== 0){
                let top = stack.at(-1)

                if(temperatures[i] <= top[0]){
                    break;
                }

                let warmDay = i - top[1]
                ans[top[1]] = warmDay

                stack.pop()
            }

            stack.push([temperatures[i], [i]])
        }

        return ans
    }
}
